src/main/haskell/kore/app/exec/Main.hs

```diff
@@ -352,16 +352,22 @@ kItemSort = groundObjectSort "SortKItem"
 -- The function below works around several limitations of
 -- the current tool by tricking the tool into believing that
 -- functions are constructors (so that function patterns can match)
--- and that @inj@ is both functional and constructor.
+-- and that @kseq@ and $dotk@ are both functional and constructor.
 constructorFunctions :: MetadataTools Object StepperAttributes -> MetadataTools Object StepperAttributes
 constructorFunctions tools =
     tools
     { symAttributes = \h -> let atts = symAttributes tools h in
         atts
-        { isConstructor = isConstructor atts || isFunction atts || isInj h
-        , isFunctional = isFunctional atts || isInj h
+        { isConstructor = isConstructor atts || isFunction atts || isCons h
+        , isFunctional = isFunctional atts || isCons h || isInj h
+        , isInjective =
+            isInjective atts || isFunction atts || isCons h || isInj h
+        , isSortInjection = isSortInjection atts || isInj h
         }
     }
   where
     isInj :: SymbolOrAlias Object -> Bool
-    isInj h = getId (symbolOrAliasConstructor h) `elem` ["inj","kseq","dotk"]
+    isInj h =
```

Comment:
Could we change the front-end to annotate `inj`, `kseq`, and `dotk` to be annotated with appropriate attributes, instead of matching the symbol name? Not _now_ necessarily, but eventually? Having certain names be special is not very nice.

Reply:
I totally agree with that. This entire `constructorFunctions` needs to disappear. They are here only temporary, until the relevant changes are effected in the frontend.